### PR TITLE
Check for bad words on challenge description

### LINKED ISSUE #31: Validate bad words
It would be good to validate bad words against the incoming challenge to before being created.
Package recomendation: https://www.npmjs.com/package/bad-words

### PR DESCRIPTION
**Pull request by namadnuno**
Closes #31

### CODE REVIEW

File: schema/ChallengeSchema.js [hunk 1]
```diff
@@ -39,7 +40,10 @@ const Challenge = mongoose.model('Challenge', challengeSchema)
 
 
 module.exports = {
-  addChallenge: (data) => new Challenge(data).save(),
+  addChallenge: (data) => {
+    data.description = sanitize(data.description);
+    return new Challenge(data).save();
+  },
```
Comment (rbgoncalves) [inline schema/ChallengeSchema.js:46]: Imo the api should throw an error when some bad word is found

File: test/challenges.test.js [hunk 2]
```diff
@@ -268,5 +268,13 @@ describe('challenge', () => {
...
         done();
       });
+    });
+
+  it('should sanitaze bad words before creating a challenge', async () => {
+    data.description = "Shit, Fuck off conas";
+    const challenge = await addChallenge(data);
+    const updatedChallenge = await findChallenge(challenge._id);
+    expect(updatedChallenge).to.be.contains(data);
+    expect(updatedChallenge.description).to.be.contains("****, **** off ***");
```
Comment (rbgoncalves) [inline test/challenges.test.js:278]: You could use `isProfane()` to make the assertion

Comment (namadnuno) [reply test/challenges.test.js]: done

File: schema/ChallengeSchema.js (hunk 1 above)
Comment (namadnuno) [reply schema/ChallengeSchema.js]: done

File: schema/ChallengeSchema.js [hunk 3]
```diff
@@ -39,7 +42,12 @@ const Challenge = mongoose.model('Challenge', challengeSchema)
 
 
 module.exports = {
-  addChallenge: (data) => new Challenge(data).save(),
+  addChallenge: (data) => {
+    if (!isProfane(data.description)) {
+      return new Challenge(data).save();
+    }
+    throw new Error('Unauthorized words have been sent');
+  },
```
Comment (rbgoncalves) [inline schema/ChallengeSchema.js:50]: I personally prefer to throw the error inside the if statement 😛

File: test/challenges.test.js [hunk 4]
```diff
@@ -269,4 +285,22 @@ describe('challenge', () => {
...
+
+  it('should return a response with an error when creating a challenge with profane words', (done) => {
+    data.description = "Shit, Fuck off conas";
+    chai.request('http://localhost:8001')
+      .post('/challenges')
+      .set('X-API-KEY', process.env.API_KEY)
+      .send(data)
+      .end(async (err, res) => {
+        expect(res).to.have.status(400);
+        expect(res.body.message).to.have.equal('Unauthorized words have been sent');
```
[the remark this answers is not in the document]
Comment (namadnuno) [reply test/challenges.test.js:302]: ok i will fix that

File: schema/ChallengeSchema.js (hunk 3 above)
Comment (namadnuno) [reply schema/ChallengeSchema.js:50]: ok i will change that

**Pull request merged**